### PR TITLE
feat: updated dependencies

### PR DESCRIPTION
**Pull request by mxschmitt**
I will fire up one single PR for bumping most of the dependencies, instead of using Dependabot for all of them.

TODO:
- [x] `node-resolve: setting options.jsnext is deprecated, please override options.mainFields instead`
- [x] reduce bundle size of the docs
- [x] improve handling of the babel configurations

### CODE REVIEW

**Comment by martijnrusschen** [comment]
Sweet, I just merged 2 PRs that seem to conflict. If you can update the PR, we can get these upgraded all in one go.

**Comment by martijnrusschen** [comment]
>  Error: Cannot find module "react-test-renderer/package.json"

**Comment by martijnrusschen** [comment]
Looks like some test are not executed somehow:
`Firefox 63.0.0 (Linux 0.0.0): Executed 242 of 406 SUCCESS (7.251 secs / 3.886 secs)`

I assume this is causing the drop in code coverage.

**Comment by martijnrusschen** [comment]
Why the build is succeeding while only executing half of the tests is a mystery to me.

**Comment by mxschmitt** [comment]
I also have no clue why not all tests were executed hmm

**Comment by mxschmitt** [comment]
@martijnrusschen fixed 🚀 
Now the rough part is done, only everything now needs to be tested 😄

File: .babelrc.js [hunk 1]
```diff
@@ -3,43 +3,28 @@ var MODULES = process.env.MODULES;
...
-        forceAllTransforms: NODE_ENV === "production"
-      }
-    ],
-    "stage-0",
-    "react"
-  ],
-  plugins: []
-};
+const presets = [
+  "airbnb",
```
Comment (mxschmitt) [inline .babelrc.js]: not sure if we need airbnb or present env or both there

File: test/exclude_time_period_test.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ describe("DatePicker", () => {
   let sandbox;
 
   beforeEach(() => {
-    sandbox = sinon.sandbox.create();
```
Comment (mxschmitt) [inline test/exclude_time_period_test.js]: was changed, because the old was was deprecated

File: .travis.yml [hunk 3]
```diff
@@ -1,7 +1,9 @@
 language: node_js
 node_js:
   - "node"
-cache: yarn
+cache:
+  - npm
```
Comment (martijnrusschen) [inline .travis.yml]: Do we use npm cache here?

File: .babelrc.js (hunk 1 above)
Comment (martijnrusschen) [reply .babelrc.js]: I don't know anymore, but if it works I'm fine with this.

Comment (martijnrusschen) [reply .babelrc.js]: Maybe remove the commented out code?

File: .travis.yml (hunk 3 above)
Comment (mxschmitt) [reply .travis.yml]: that will add node_modules to the cache. (but since july 2019 it's default active)

**Comment by mxschmitt** [comment]
> Do you feel comfortable merging as is?

not sure I've got one issue left, reduce bundle size of the docs js bundle.

**Comment by mxschmitt** [comment]
I guess we are ready to merge. But I have not tested yet the different output files which will distributed over NPM.

**Comment by martijnrusschen** [comment]
It seems the docs site is down since merging this PR - https://reactdatepicker.com/

**Comment by mxschmitt** [comment]
See #1860 for the fix.

**Comment by martijnrusschen** [comment]
Cool!

**Pull request merged**